src/contracts/Value/ItemListInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Personalization\Value;
+
+use Closure;
+use Countable;
+use IteratorAggregate;
+
+/**
+ * @extends IteratorAggregate<ItemInterface>
+ */
+interface ItemListInterface extends IteratorAggregate, Countable
+{
+    /**
+     * @throws \EzSystems\EzRecommendationClient\Exception\ItemNotFoundException
+     */
+    public function get(string $identifier, string $language): ItemInterface;
+
+    public function has(string $identifier, string $language): bool;
+
+    public function filter(Closure $predicate): self;
```

Comment:
```suggestion
    public function filter(callable $predicate): self;
```

You might want to declare in phpdoc that this method will create & return a second instance. That's usually how selectable collections work.

See https://github.com/doctrine/collections/blob/b33e58d43b12ca7cc4d14f3b17849c517fd77059/lib/Doctrine/Common/Collections/ArrayCollection.php#L358

And don't forget to describe the callable: what arguments will it receive and on which positions.